@xen-orchestra/fs/src/s3.js

```diff
@@ -32,9 +32,10 @@ export default class S3Handler extends RemoteHandlerAbstract {
     if (protocol === 'http') {
       params.httpOptions.agent = new http.Agent()
       params.sslEnabled = false
-    } else if (protocol === 'https' && allowUnauthorized === true) {
+    } else if (protocol === 'https') {
       params.httpOptions.agent = new https.Agent({
-        rejectUnauthorized: false,
+        rejectUnauthorized: !allowUnauthorized,
+        keepAlive: true,
```

Comment:
Should we add it to HTTP as well?

Reply:
yes. I don't think that will have the same impact, since there is no TLS negociation, but let's try to be symmetric